Lib/test/test_pyexpat.py

```diff
@@ -821,5 +823,169 @@ def start_element(name, _):
+        raise NotImplementedError
+
+    def test_set_attack_protection_threshold_ignored(self):
+        raise NotImplementedError
+
+    def test_set_attack_protection_threshold_arg_invalid_type(self):
+        parser = expat.ParserCreate()
+        setter = functools.partial(self.set_activation_threshold, parser)
+
+        self.assertRaises(TypeError, setter, 1.0)
+        self.assertRaises(TypeError, setter, -1.5)
+        self.assertRaises(ValueError, setter, -5)
+
+    def test_set_attack_protection_threshold_arg_invalid_range(self):
+        _testcapi = import_helper.import_module("_testcapi")
+        parser = expat.ParserCreate()
+        setter = functools.partial(self.set_activation_threshold, parser)
+
+        self.assertRaises(OverflowError, setter, _testcapi.ULLONG_MAX + 1)
+
+    def test_set_attack_protection_threshold_fail_for_subparser(self):
+        parser = expat.ParserCreate()
+        subparser = parser.ExternalEntityParserCreate(None)
+        setter = functools.partial(self.set_activation_threshold, subparser)
+        self.assert_root_parser_failure(setter, 12345)
+
+    def test_set_maximum_amplification_reached(self):
+        raise NotImplementedError
+
+    def test_set_maximum_amplification_ignored(self):
+        raise NotImplementedError
+
+    def test_set_maximum_amplification_infinity(self):
+        inf = float('inf')  # an 'inf' threshold is allowed by Expat
+        parser = expat.ParserCreate()
+        self.assertIsNone(self.set_maximum_amplification(parser, inf))
+
+    def test_set_maximum_amplification_arg_invalid_type(self):
+        parser = expat.ParserCreate()
+        setter = functools.partial(self.set_maximum_amplification, parser)
+
+        self.assertRaises(TypeError, setter, None)
+        self.assertRaises(TypeError, setter, 'abc')
+
+    def test_set_maximum_amplification_arg_invalid_range(self):
+        parser = expat.ParserCreate()
+        setter = functools.partial(self.set_maximum_amplification, parser)
+
+        msg = re.escape("'max_factor' must be at least 1.0")
+        self.assertRaisesRegex(expat.ExpatError, msg, setter, float('nan'))
+        self.assertRaisesRegex(expat.ExpatError, msg, setter, 0.99)
+
+    def test_set_maximum_amplification_fail_for_subparser(self):
+        parser = expat.ParserCreate()
+        subparser = parser.ExternalEntityParserCreate(None)
+        setter = functools.partial(self.set_maximum_amplification, subparser)
+        self.assert_root_parser_failure(setter, 123.45)
+
+
+@unittest.skipIf(expat.version_info < (2, 7, 2), "requires Expat >= 2.7.2")
+class MemoryProtectionTest(AttackProtectionTestCases, unittest.TestCase):
+
+    # With the default Expat configuration, the billion laughs protection may
+    # hit before the allocation limiter if exponential_expansion_payload() is
+    # not carefully parametrized. In particular, use the following assert_*()
+    # methods to check the error message of the active protection.
```

Comment:
```suggestion

    # NOTE:
    # With the default Expat configuration, the billion laughs protection may
    # hit before the allocation limiter if exponential_expansion_payload() is
    # not carefully parametrized. In particular, use the following assert_*()
    # methods to check the error message of the active protection.
```
? (could use more visibility)

Reply:
I think the comment indicates sufficiently well that it is a note-like comment.

Reply:
@picnixz yes, my point was that it's not loud enough. That I tried to say with "could use more visibility". It's too easy to miss. Just my two cents.